xbmc/utils/SysfsUtils.h

```diff
@@ -10,13 +10,21 @@
 
 #include <string>
 
-class SysfsUtils
+class CSysfs
 {
 public:
-  static int SetString(const std::string& path, const std::string& valstr);
-  static int GetString(const std::string& path, std::string& valstr);
-  static int SetInt(const std::string& path, const int val);
-  static int GetInt(const std::string& path, int& val);
-  static bool Has(const std::string& path);
-  static bool HasRW(const std::string &path);
+  CSysfs() = default;
+  CSysfs(const std::string& path) : m_path(path) {}
+  ~CSysfs() = default;
+
+  void operator=(const std::string& path) { m_path = path; }
+
+  template<typename T>
+  T Get();
```

Comment:
Not too happy with this to be honest. It is not clear which types can be used from the header. If you try `Get<unsigned int>()` or `Get<short>()` you will just get a linker error (since it would not use the `<int>` specialization and there is none defined for `<unsigned int>`), not too helpful. And it is not generic either, you have an explicit specialization for each type.

My suggestion would be that you make it actually generic (so, for example, support at least all arithmetic types) or use the previous approach of distinct functions which would at least allow for automatic conversion of integers.

Same for `Set()`.